### PR TITLE
feat(SpeakerLayout): add participantsBarLimit

### PR DESCRIPTION
**Pull request by arnautov-anton**
Adds `participantsBarLimit` property to `SpeakerLayout` component which hard-limits number of rendered participants in the participants bar.

### CODE REVIEW

File: packages/react-sdk/src/core/hooks/useCalculateHardLimit.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    if (
+      !hostElement ||
+      !wrapperElement ||
+      typeof limit === 'number' ||
+      typeof limit === 'undefined'
+    )
+      return;
+
+    let childWidth = 280;
+    let childHeight = 160;
```
Comment (oliverlaz) [inline packages/react-sdk/src/core/hooks/useCalculateHardLimit.ts:32]: Leftovers from testing?

File: packages/react-sdk/src/core/components/CallLayout/SpeakerLayout.tsx [hunk 2]
```diff
@@ -32,6 +33,11 @@ export type SpeakerLayoutProps = {
    * Providing `null` will hide the bar.
    */
   participantsBarPosition?: 'top' | 'bottom' | 'left' | 'right' | null;
+  /**
+   * Hard limits the number of the participants rendered in the participants bar.
+   * Providing string `dynamic` will calculate hard limit based on screen width/height.
+   */
+  participantsBarLimit?: 'dynamic' | number;
```
Comment (oliverlaz) [inline packages/react-sdk/src/core/components/CallLayout/SpeakerLayout.tsx:40]: Why not make this an optional number? I think it is already used like that. By default it is `undefined` but when provided, we use the value. I don't think we need a `dynamic` as a value here.

Comment (arnautov-anton) [reply packages/react-sdk/src/core/components/CallLayout/SpeakerLayout.tsx:40]: The functionality can reach three states - without the limit (undefined) - it's used with the normal - scrolling behaviour where no limitation is applied, number to limit it to that specific amount and "dynamic" to calculate the amount based on the screen width/height which will render maximal amount that fits into the parent and does not trigger scrolling behavior. Optional number would mean to limit it only to two of the aforemention behaviours.

File: packages/react-sdk/src/core/hooks/useCalculateHardLimit.ts (hunk 1 above)
Comment (arnautov-anton) [reply packages/react-sdk/src/core/hooks/useCalculateHardLimit.ts:32]: Defaults to allow at least one element to render to get proper values for future calculations but now looking at it I'd rather do `calculatedLimit || 1` in the return statement.

Comment (arnautov-anton) [reply packages/react-sdk/src/core/hooks/useCalculateHardLimit.ts]: Resolved here: https://github.com/GetStream/stream-video-js/pull/1090/commits/6537678ace8f208f30acce565cc1cfe4b76be864

**Review by oliverlaz — APPROVED** [review]

**Pull request merged**